app/scripts/modules/core/account/account.service.ts

```diff
@@ -0,0 +1,210 @@
+import {chain, intersection, memoize} from 'lodash';
+import {module} from 'angular';
+
+import {Application} from 'core/application/application.model';
+import {API_SERVICE, Api} from 'core/api/api.service';
+
+export interface IRegion {
+  name: string;
+  preferredZones: string[];
+  availablityZones: string[];
+}
+
+export interface IAccount {
+  accountId: string;
+  name: string;
+  regions: IRegion[];
+  requiredGroupMembership: string[];
+  type: string;
+}
+
+export interface IZone {
+  [key: string]: string[];
+}
+
+export interface IAccountZone {
+  [key: string]: IZone;
+}
+
+export class AccountService {
+
+  static get $inject() {
+    return ['$log', '$q', 'settings', 'infrastructureCaches', 'cloudProviderRegistry', 'API'];
+  }
+
+  constructor(private $log: ng.ILogService,
+              private $q: ng.IQService,
+              private settings: any,
+              private infrastructureCaches: any,
+              private cloudProviderRegistry: any,
+              private API: Api) {
+  }
+
+  public challengeDestructiveActions(account: string): ng.IPromise<any> {
```

Comment:
can `ng.IPromise<any>` be `ng.IPromise<boolean>` here? Probably need to cast `details.challengeDestructiveActions` on L50 and `ng.IQResolveReject<any>` on L45 to booleans but seems like it'd be a little cleaner.